src/lib/Runner.ts

```diff
@@ -630,6 +636,57 @@ export class Runner {
     );
   };
 
+  // this check prevents the system from being hung if BB webhook doesn't work as expected
+  checkRunningLandRequests = async () => {
+    await withLock(
+      // this lock ensures we don't run multiple checks at the same time that might cause a race condition
+      'check-running-requests',
+      async () => {
+        await withLock(
+          // this lock ensures we don't run the check when we're running this.next()
+          'status-transition',
+          async () => {
+            const requests = await this.queue.getRunning();
+            const runningRequests = requests.filter((request) => request.state === 'running');
+
+            Logger.info('Checking running landrequests for timeout', {
+              namespace: 'lib:runner:checkRunningLandRequests',
+              runningRequests,
+            });
+
+            for (const landRequestStatus of runningRequests) {
+              const timeElapsed = Date.now() - landRequestStatus.date.getTime();
+
+              if (timeElapsed > LAND_BUILD_TIMEOUT_TIME) {
+                const landRequest = landRequestStatus.request;
+
+                Logger.warn('Failing running land request as timeout period is breached', {
+                  pullRequestId: landRequest.pullRequestId,
+                  landRequestId: landRequest.id,
+                  namespace: 'lib:runner:checkRunningLandRequests',
+                });
+                await landRequest.setStatus('fail', 'Build timeout period breached');
+              } else {
+                const { buildId } = landRequestStatus.request;
+                const { state } = await this.client.getLandBuild(buildId);
+
+                // buildStatus can be SUCCESSFUL, FAILED or STOPPED
+                this.onStatusUpdate({
+                  buildId,
+                  buildStatus: state.result?.name,
+                });
```

Comment:
I think we should be `await`ing the `onStatusUpdate` call since its async

Reply:
Yeah, I realised this from the logging as well while testing. Added the change.